src/rollup-db.js

```diff
@@ -187,17 +189,30 @@ class RollupDB {
      * @returns {Object} Exit tree information
      */
     async getExitInfo(idx, numBatch){
-        const stateRoot = await this.getStateRoot(numBatch);
-
-        if (!stateRoot) return null;
+        let merkleRoot;
+        if (numBatch > this.migrationBatch) {
+            merkleRoot = await this.getStateRoot(numBatch);
+        } else { 
+            // use legacy exit tree
+            const rootValues = await this.getStateRoot(numBatch);
+            merkleRoot = rootValues[1];
+        }
+        if (!merkleRoot) return null;
+        
         const dbState = new SMTTmpDb(this.db);
-        const tmpStateTree = new SMT(dbState, stateRoot);
+        const tmpStateTree = new SMT(dbState, merkleRoot);
```

Comment:
if legacy is retrieved then is not the `stateTree` but the exit one.
Maybe is better to code a function like `getExitInfoLegacy` ?

Reply:
ok i will do that ^^